kong/plugins/aws-lambda/iam-eks-credentials.lua

```diff
@@ -0,0 +1,77 @@
+local kong = kong
+local ENV_TOKEN_FILE = os.getenv 'AWS_WEB_IDENTITY_TOKEN_FILE'
+local ENV_ROLE_ARN = os.getenv 'AWS_ROLE_ARN'
+local DEFAULT_SERVICE_REQUEST_TIMEOUT = 5000
+
+
+local url = require "socket.url"
+local http = require "resty.http"
+local json = require "cjson"
+local ngx_now = ngx.now
+local concat = table.concat
+local tostring = tostring
+
+do
+  if not (ENV_TOKEN_FILE) then
+    -- No variables found, so we're not probably not running on EKS containers
+    kong.log.debug("No Web Identity environment variables found for IAM")
+  end
+end
+
+
+local function fetchCredentials()
```

Comment:
Do we expect the contents `ENV_TOKEN_FILE` to change? If not, then it's possible to just read it once and cache the result in a module-level var.

```lua
local pl_file = require "pl.file"
local token

local function fetchCredentials()
  if token == nil then
    local content, err = pl_file.read(ENV_TOKEN_FILE)
    token = content or false
    if not token then
      kong.debug.warn("failed reading ", ENV_TOKEN_FILE, ": ", err)
    end      
  end
  
  if not token then
    return nil, "no token"
  end
  
  local client = http.new()
  client:set_timeout(DEFAULT_SERVICE_REQUEST_TIMEOUT)

  -- [...]
end
```

And if we _do_ need to re-read `ENV_TOKEN_FILE` every time we're called, it's probably still better to move the file error-handling up top so that we're not creating a new http client if we don't have a token for it.